### PR TITLE
[CP-895] I want to be notify if I have new message while message thread is open

### PR DESCRIPTION
**Pull request by wercix**
Jira: [CP-895]

**Description**

<details>
<summary><b>Screenshots</b></summary>

Screen Shot 2022-05-09 at 12 42 34

</details>

[CP-895]: https://appnroll.atlassian.net/browse/CP-895?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/messages/components/thread-details-messages.component.tsx [hunk 1]
```diff
@@ -58,6 +157,7 @@ const ThreadDetailsMessages: FunctionComponent<Properties> = ({
           return <MessageDayBubble key={id} {...messageDayBubble} />
         }}
       </ViewportList>
+      <div ref={newRef}></div>
```
Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:160]: What means `newRef` 😉 New relative to another instance? What next if will occur the new one? 😉 
```suggestion
      <div ref={wrapperBottomRef} />
```

File: packages/app/src/messages/components/thread-details-messages.component.tsx [hunk 2]
```diff
@@ -14,25 +20,118 @@ import {
...
+  }
+
+  let observer: IntersectionObserver
+
+  const callback = (entries: IntersectionObserverEntry[]) => {
+    if (entries[0] && !entries[0].isIntersecting) {
+      return
+    } else if (entries[0] && entries[0].isIntersecting) {
+      closeNewMessageBadge()
+    }
```
Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:90]:
```suggestion
  if (entries[0] && entries[0].isIntersecting) {
      closeNewMessageBadge()
    }
```

Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:90]:
```suggestion
  if (entries[0] && entries[0].isIntersecting) {
      closeNewMessageBadge()
    }
```

File: packages/app/src/messages/components/thread-details-messages.component.tsx [hunk 3]
```diff
@@ -14,25 +20,118 @@ import {
...
   receiver,
+  messageLayoutNotifications,
+  removeLayoutNotification,
 }) => {
-  const ref = useRef(null);
+  const newRef = useRef<HTMLDivElement>(null)
+  const ref = useRef<HTMLDivElement>(null)
+  const [notifications, setNotifications] = useState<
+    Notification[] | undefined
+  >()
```
Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:43]: Isn't be easier to handle logic  just with array type? 🤔 
```suggestion
  const [notifications, setNotifications] = useState<
    Notification[]
  >([])
```

Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:43]: Isn't be easier to handle logic  just with array type? 🤔 
```suggestion
  const [notifications, setNotifications] = useState<
    Notification[]
  >([])
```

File: packages/app/src/messages/components/thread-details-messages.component.tsx (hunk 2 above)
Comment (wercix) [reply packages/app/src/messages/components/thread-details-messages.component.tsx]: Fixed on [aef71f2](https://github.com/mudita/mudita-center/pull/712/commits/aef71f29194ecd6b1ab1e0ef09ab5ffd06f447ac)

File: packages/app/src/messages/components/thread-details-messages.component.tsx (hunk 1 above)
Comment (wercix) [reply packages/app/src/messages/components/thread-details-messages.component.tsx]: Fixed on [aef71f2](https://github.com/mudita/mudita-center/pull/712/commits/aef71f29194ecd6b1ab1e0ef09ab5ffd06f447ac)

File: packages/app/src/messages/components/thread-details-messages.component.tsx [hunk 4]
```diff
@@ -14,25 +20,113 @@ import {
...
+    return () => {
+      if (wrapperBottomRef.current) {
+        observer.unobserve(wrapperBottomRef.current)
+      }
+    }
+  }, [wrapperBottomRef, messageLayoutNotifications])
 
   return (
     <MessageBubblesWrapper ref={ref}>
+      {notifications && notifications.length > 0 && (
```
Comment (dkarski) [inline packages/app/src/messages/components/thread-details-messages.component.tsx:116]:
```suggestion
      {notifications.length > 0 && (
```

File: packages/app/src/messages/components/thread-details-messages.component.tsx (hunk 3 above)
Comment (dkarski) [reply packages/app/src/messages/components/thread-details-messages.component.tsx]: no complete 😉 ➡️ https://github.com/mudita/mudita-center/pull/712/files#r869989545

File: packages/app/src/messages/components/thread-details-messages.component.tsx (hunk 4 above)
Comment (wercix) [reply packages/app/src/messages/components/thread-details-messages.component.tsx]: Fixed on [bd6a8e1](https://github.com/mudita/mudita-center/pull/712/commits/bd6a8e1d55ae99d73ce4d756240c62a1f3362711)

File: packages/app/src/messages/components/thread-details-messages.component.tsx (hunk 3 above)
Comment (wercix) [reply packages/app/src/messages/components/thread-details-messages.component.tsx]: You're right! Fixed on [bd6a8e1](https://github.com/mudita/mudita-center/pull/712/commits/bd6a8e1d55ae99d73ce4d756240c62a1f3362711)

**Review by dkarski — APPROVED** [review]

**Review by igorbezsmertnyi — APPROVED** [review]

**Pull request merged**